File: src/lib/pjecalc/core/dominio/calculoexterno/parcelas-atualizaveis-honorario-utils.ts
/**
 * PJe-Calc v2.15.1 — ParcelasAtualizaveisHonorarioUtils
 * Porte de: br.jus.trt8.pjecalc.negocio.dominio.calculoexterno.ParcelasAtualizaveisHonorarioUtils
 */
import Decimal from 'decimal.js';
import { MensagemDeRecurso } from '../../comum/mensagem-de-recurso';
import { Mensagens } from '../../comum/mensagens';
import { TipoValorEnum } from '../../constantes/enums';
import { ParcelasAtualizaveisHonorario } from './parcelas-atualizaveis-honorario';
import { ParcelasAtualizaveisUtils } from './parcelas-atualizaveis-utils';

const ZERO = new Decimal(0);

export class ParcelasAtualizaveisHonorarioUtils extends ParcelasAtualizaveisUtils {
  static consistirDados(pa: ParcelasAtualizaveisHonorario): void {
    if (!pa.getApurarIrpf()) {
      pa.setIncidirIrpfSobreJuros(false);
      pa.setTipoIrpf(null);
    }
    if (pa.getTipoValor() === TipoValorEnum.INFORMADO) {
      pa.setAplicarDescontoContribSocialCalculado(false);
      pa.setAplicarDescontoPrevPrivadaCalculado(false);
      pa.setTaxaCalculado(null);
      pa.setValorParcelaInformado(pa.getValorParcelaInformado() ?? ZERO);
      pa.setValorJurosInformado(pa.getValorJurosInformado() ?? ZERO);
    } else {
      pa.setValorParcelaInformado(null);
      pa.setValorJurosInformado(null);
      pa.setIndiceTrabalhistaInformado(null);
      pa.setAplicarJurosInformado(false);
      pa.setDataApartirDeAplicarJurosInformado(null);
    }
  }

  static validarPreenchimentoFormulario(
    paHonorario: ParcelasAtualizaveisHonorario,
    idDescricao: string,
    idCredor: string | null,
    idValorParcela: string,
    idAliquota: string,
    idNumDocFiscal: string,
    idValorJuros: string,
  ): MensagemDeRecurso[] {
    const erros: MensagemDeRecurso[] = [];
    const descricao = paHonorario.getDescricao();
    if (!descricao || descricao.trim() === '') {
      erros.push(new MensagemDeRecurso(idDescricao, Mensagens.MSG0003, 'Descrição'));
    }
    if (idCredor !== null) {
      const credor = paHonorario.getCredor();
      if (!credor || credor.trim() === '') {
        erros.push(new MensagemDeRecurso(idCredor, Mensagens.MSG0003, 'Credor'));
      }
    }
    if (paHonorario.getTipoValor() === TipoValorEnum.CALCULADO) {
      if (paHonorario.getTaxaCalculado() == null) {
        erros.push(new MensagemDeRecurso(idAliquota, Mensagens.MSG0003, 'Alíquota'));
      }
    } else {
      if (paHonorario.getValorParcelaInformado() == null) {
        erros.push(new MensagemDeRecurso(idValorParcela, Mensagens.MSG0003, 'Valor da Parcela'));
      }
      const valorJuros = paHonorario.getValorJurosInformado();
      if (
        paHonorario.getAplicarJurosInformado() &&
        paHonorario.getDataApartirDeAplicarJurosInformado() != null &&
        valorJuros != null &&
        !valorJuros.equals(ZERO)
      ) {
        erros.push(new MensagemDeRecurso(idValorJuros, Mensagens.MSG0196));
      }
    }
    const numeroDocFiscal = paHonorario.getNumeroDocFiscal();
    if (paHonorario.getApurarIrpf() && (!numeroDocFiscal || numeroDocFiscal.trim() === '')) {
      erros.push(new MensagemDeRecurso(idNumDocFiscal, Mensagens.MSG0003, 'Número'));
    }
    // TODO(fase-15): GerenciadorDeValidadores.getInstance().validar(...)
    return erros;
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculoexterno/ParcelasAtualizaveisHonorarioUtils.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.apache.commons.lang.StringUtils
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculoexterno;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.comum.GerenciadorDeValidadores;
import br.jus.trt8.pjecalc.negocio.comum.MensagemDeRecurso;
import br.jus.trt8.pjecalc.negocio.constantes.Mensagens;
import br.jus.trt8.pjecalc.negocio.constantes.TipoValorEnum;
import br.jus.trt8.pjecalc.negocio.dominio.calculoexterno.ParcelasAtualizaveisHonorario;
import br.jus.trt8.pjecalc.negocio.dominio.calculoexterno.ParcelasAtualizaveisUtils;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import org.apache.commons.lang.StringUtils;

public class ParcelasAtualizaveisHonorarioUtils
extends ParcelasAtualizaveisUtils {
    public static void consistirDados(ParcelasAtualizaveisHonorario parcelasAtualizaveisHonorario) {
        if (!parcelasAtualizaveisHonorario.getApurarIrpf().booleanValue()) {
            parcelasAtualizaveisHonorario.setIncidirIrpfSobreJuros(false);
            parcelasAtualizaveisHonorario.setTipoIrpf(null);
        }
        if (parcelasAtualizaveisHonorario.getTipoValor().equals((Object)TipoValorEnum.INFORMADO)) {
            parcelasAtualizaveisHonorario.setAplicarDescontoContribSocialCalculado(false);
            parcelasAtualizaveisHonorario.setAplicarDescontoPrevPrivadaCalculado(false);
            parcelasAtualizaveisHonorario.setTaxaCalculado(null);
            parcelasAtualizaveisHonorario.setValorParcelaInformado(parcelasAtualizaveisHonorario.getValorParcelaInformado() == null ? BigDecimal.ZERO : parcelasAtualizaveisHonorario.getValorParcelaInformado());
            parcelasAtualizaveisHonorario.setValorJurosInformado(parcelasAtualizaveisHonorario.getValorJurosInformado() == null ? BigDecimal.ZERO : parcelasAtualizaveisHonorario.getValorJurosInformado());
        } else {
            parcelasAtualizaveisHonorario.setValorParcelaInformado(null);
            parcelasAtualizaveisHonorario.setValorJurosInformado(null);
            parcelasAtualizaveisHonorario.setIndiceTrabalhistaInformado(null);
            parcelasAtualizaveisHonorario.setAplicarJurosInformado(false);
            parcelasAtualizaveisHonorario.setDataApartirDeAplicarJurosInformado(null);
        }
    }

    public static List<MensagemDeRecurso> validarPreenchimentoFormulario(ParcelasAtualizaveisHonorario paHonorario, String idDescricao, String idCredor, String idValorParcela, String idAliquota, String idNumDocFiscal, String idValorJuros) {
        ArrayList<MensagemDeRecurso> erros = new ArrayList<MensagemDeRecurso>();
        if (StringUtils.isBlank((String)paHonorario.getDescricao())) {
            erros.add(new MensagemDeRecurso(idDescricao, Mensagens.MSG0003, "Descri\u00e7\u00e3o"));
        }
        if (idCredor != null && StringUtils.isBlank((String)paHonorario.getCredor())) {
            erros.add(new MensagemDeRecurso(idCredor, Mensagens.MSG0003, "Credor"));
        }
        if (paHonorario.getTipoValor().equals((Object)TipoValorEnum.CALCULADO)) {
            if (paHonorario.getTaxaCalculado() == null) {
                erros.add(new MensagemDeRecurso(idAliquota, Mensagens.MSG0003, "Al\u00edquota"));
            }
        } else {
            if (paHonorario.getValorParcelaInformado() == null) {
                erros.add(new MensagemDeRecurso(idValorParcela, Mensagens.MSG0003, "Valor da Parcela"));
            }
            if (paHonorario.getAplicarJurosInformado().booleanValue() && Utils.naoNulo(paHonorario.getDataApartirDeAplicarJurosInformado()) && Utils.naoNulo(paHonorario.getValorJurosInformado()) && BigDecimal.ZERO.compareTo(paHonorario.getValorJurosInformado()) != 0) {
                erros.add(new MensagemDeRecurso(idValorJuros, Mensagens.MSG0196, new Object[0]));
            }
        }
        if (paHonorario.getApurarIrpf().booleanValue() && StringUtils.isBlank((String)paHonorario.getNumeroDocFiscal())) {
            erros.add(new MensagemDeRecurso(idNumDocFiscal, Mensagens.MSG0003, "N\u00famero"));
        }
        GerenciadorDeValidadores.getInstance().validar(ParcelasAtualizaveisHonorario.class, paHonorario);
        return erros;
    }
}

